sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/package.scala

```diff
@@ -42,6 +42,10 @@ package object analysis {
       throw new AnalysisException(msg, t.origin.line, t.origin.startPosition)
     }
 
+    def failAnalysis(error: Throwable): Nothing = {
```

Comment:
is this method necessary? Can we just let the caller side use throw directly?

Reply:
done